File: 04-Striver/1-Rcursion/programs/2-print-finite-recursion.py
# print recursion function
# With base case
# Finite times
count = 0
class solution:
    def recursion(self):
        global count
        if count == 3:
            return
        print(count)
        count += 1
        self.recursion()
    def __init__(self):
        self.recursion()

test = solution()

File: 03-Take-U-Forward/1-Recursion/programs/2-print-finite-recursion.cpp
// print recursion function finite times
// With base condition
#include<bits/stdc++.h>
using namespace std;

int count = 0;
void print()
{
    if (count == 3)
        return ;
    cout<<count<<endl;
    count++;
    print();
}
int main()
{
    // #ifdef ONLINE_JUDGE
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "r", stdout);
    // #endif
    print();
    return 0;
}
